activesupport/lib/active_support/hash_with_indifferent_access.rb

```diff
@@ -195,6 +195,19 @@ def values_at(*indices)
       indices.collect { |key| self[convert_key(key)] }
     end
 
+    # Returns an array of the values at the specified indices, but also
+    # raises an exception when one of the keys can't be found.
+    #
+    #   hash = ActiveSupport::HashWithIndifferentAccess.new
+    #   hash[:a] = 'x'
+    #   hash[:b] = 'y'
+    #   hash.fetch_values('a', 'b') # => ["x", "y"]
+    #   hash.fetch_values('a', 'c') { |key| 'z' } # => ["x", "z"]
+    #   hash.fetch_values('a', 'c') # => KeyError: key not found: "c"
+    def fetch_values(*indices, &block)
+      indices.collect { |key| fetch(key, &block) }
+    end
```

Comment:
Let's throw an `if Hash.method_defined? :fetch_values` onto the `end` here, so we don't get ahead of Hash.

(And a skip in the test to match)